test/test_signal.cc

```diff
@@ -0,0 +1,14 @@
+#include "glog/logging.h"
+#include "gtest/gtest.h"
+
+void f(int i) {
+  if (i >= 10) {
+    CHECK(false);
+  } else {
+    f(i + 1);
+  }
+};
+
+TEST(SignalTestSTD, CheckFail) {
+  f(20);
```

Comment:
Document what this test is supposed to do.

It looks like this test will fail, which looks undesirable. Do we want something like `EXPECT_DEATH` to ensure that this crashes with good output?

If that doesn't work, maybe we want this test to run a subprocess which crashes, then we could inspect its log output to make sure that it printed a good stack trace. Example of running a subprocess that we expect to crash: https://github.com/3rdparty/eventuals/blob/main/test/grpc/client-death-test.cc#L64